core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -506,6 +506,7 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     val (sched, ts) = SparkContext.createTaskScheduler(this, master)
     _schedulerBackend = sched
     _taskScheduler = ts
+    logDebug("Starting DAGScheduler")
```

Comment:
I'd rather move it to DAGScheduler's constructor as that's where the message belongs.